messaging/propagate.go

```diff
@@ -201,21 +201,18 @@ func (p *Propagate) Dispatch() error {
 				if err := p.SendToParent(&PropagateReply{}); err != nil {
 					return err
 				}
-			}
-			if p.IsLeaf() {
 				process = false
-			} else {
-				log.Lvl3(p.ServerIdentity(), "Sending to children")
-				if errs = p.SendToChildrenInParallel(&msg.PropagateSendData); len(errs) != 0 {
-					var errsStr []string
-					for _, e := range errs {
-						errsStr = append(errsStr, e.Error())
-					}
-					if len(errs) > p.allowedFailures {
-						return errors.New(strings.Join(errsStr, "\n"))
-					}
-					log.Lvl2("Error while sending to children:", errsStr)
+			}
+			log.Lvl3(p.ServerIdentity(), "Sending to children", p.Children())
+			if errs = p.SendToChildrenInParallel(&msg.PropagateSendData); len(errs) != 0 {
+				var errsStr []string
+				for _, e := range errs {
+					errsStr = append(errsStr, e.Error())
+				}
+				if len(errs) > p.allowedFailures {
```

Comment:
You can move the check earlier, when checking for `len(errs) != 0`; but that depends on what you want to log.

Reply:
Yes, that whole thing is quite crappy anyway: it was supposed to be a fast propagation protocol in a tree-like structure. But then it failed when an intermediate node was down, so I flattened the tree... Now it has lots of stuff that is not really needed anymore.

But, yes, I'd like to log any error I get in there.